dataset/src/main/scala/frameless/functions/AggregateFunctions.scala

```diff
@@ -40,24 +40,24 @@ trait AggregateFunctions {
     new TypedAggregateAndColumn[T, Out, Option[Out]](untyped.avg(column.untyped))
   }
 
-  def variance[A: Variance, T](column: TypedColumn[T, A]): TypedAggregateAndColumn[T, A, Option[A]] = {
-    import column.uencoder
-    new TypedAggregateAndColumn[T, A, Option[A]](untyped.variance(column.untyped))
+  // In Spark variance always returns Double
+  // https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala#186
```

Comment:
Might want to use the [canonical url](https://github.com/apache/spark/blob/4a3c09601ba69f7d49d1946bb6f20f5cfe453031/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala#186) istead, or line number will get outdated pretty quickly. (BTW if you type `Y` anywhere on GH it changes the current url to it's canonical form)